File: server/random/AbstractRandom.ts

export abstract class AbstractRandom {

    abstract generateInt(min: number, max: number): number;

    public generateList(entries: number, min: number, max: number): Array<number> {
        const list: Array<number> = [];
        while (list.length < entries) {
            const value = this.generateInt(min, max);
            if (list.includes(value) === false) {
                list.push(value);
            }
        }
        return list.sort((a, b) => a - b);
    }
}

File: java-server/src/main/java/com/random/AbstractRandom.java
package com.random;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public abstract class AbstractRandom {
    public abstract int generateInt(int min, int max);

    public List<Integer> generateList(int entries, int min, int max) {
        List<Integer> list = new ArrayList<>();
        while (list.size() < entries) {
            int value = this.generateInt(min, max);
            if (list.contains(value) == false) {
                list.add(value);
            }
        }
        Collections.sort(list);
        return list;
    }
}
